x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -70,3 +193,59 @@ func createPoolRestriction(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
 		return hasTwoCoins && hasPoolCreationFee
 	}
 }
+
+// getRandCLPool gets a concnerated liquidity pool with its pool denoms.
+func getRandCLPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (cltypes.ConcentratedPoolExtension, poolmanagertypes.PoolI, []string, error) {
+	// get all pools
+	clPools, err := k.GetAllPools(ctx)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	numPools := len(clPools)
+	if numPools == 0 {
+		return nil, nil, nil, fmt.Errorf("no pools created")
+	}
+
+	pool_id := clPools[rand.Intn(numPools)].GetId()
+
+	// check if the pool exists
+	poolI, err := k.GetPool(ctx, pool_id)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("Pool not found for id %d", pool_id)
+	}
+
+	concentratedPool, ok := poolI.(cltypes.ConcentratedPoolExtension)
+	if !ok {
+		return nil, nil, nil, fmt.Errorf("interface conversion failed")
+	}
+
+	poolDenoms := []string{concentratedPool.GetToken0(), concentratedPool.GetToken1()}
+
+	return concentratedPool, poolI, poolDenoms, err
+}
+
+// getRandomTickPositions returns random lowerTick and upperTick divisible by tickSpacing value.
+func getRandomTickPositions(minTick, maxTick int64, tickSpacing uint64) (int64, int64, error) {
+	lower_tick_disivisble := ((maxTick - minTick) / int64(tickSpacing)) + 1 // get random value between minTick, maxTick that is divisible by TickSpacing
+	if lower_tick_disivisble < 1 {
+		return 0, 0, fmt.Errorf("lower tick divisible by tickspacing not found")
+	}
```

Comment:
Might want to rewrite / add more comments here, it took me a while to figure out how it was working.